ui/src/app/views/cohort-edit/component.spec.ts

```diff
@@ -0,0 +1,267 @@
+import {TestBed, async, fakeAsync, tick} from '@angular/core/testing';
+import {Title, By} from '@angular/platform-browser';
+import {FormsModule} from '@angular/forms';
+import {ActivatedRoute, UrlSegment} from '@angular/router';
+import {CohortEditComponent} from 'app/views/cohort-edit/component';
+import {RouterTestingModule} from '@angular/router/testing';
+import {BehaviorSubject} from 'rxjs/BehaviorSubject';
+import {SignInService} from 'app/services/sign-in.service';
+import {WorkspaceComponent} from 'app/views/workspace/component';
+
+import {Cohort, CohortListResponse} from 'generated';
+import {CohortsService} from 'generated';
+import { Observable } from 'rxjs/Observable';
+
+class Workspace {
+  id: string;
+  namespace: string;
+  cohorts: Cohort[];
+}
+
+function simulateInput(fixture: any, element: any, text: string) {
+  element.value = text;
+  element.dispatchEvent(new Event('input'));
+  updateAndTick(fixture);
+}
+
+function updateAndTick(fixture: any) {
+  fixture.detectChanges();
+  tick();
+}
+
+class CohortsServiceStub {
+  constructor() {
+    const stubWorkspace = new Workspace();
+    stubWorkspace.id = WorkspaceComponent.DEFAULT_WORKSPACE_ID;
+    stubWorkspace.namespace = WorkspaceComponent.DEFAULT_WORKSPACE_NS;
+
+    const exampleCohort: Cohort = {id: '',
+      name: '', description: '', criteria: '', type: ''};
+    exampleCohort.id = '1';
+    exampleCohort.name = 'sample name';
+    exampleCohort.description = 'sample description';
+
+    stubWorkspace.cohorts = [exampleCohort];
+    this.workspaces = [stubWorkspace];
+  }
+  public workspaces: Workspace[];
+
+  private getMatchingWorkspace(wsNamespace: string, wsId: string): Workspace {
+    return this.workspaces.find(function(workspace: Workspace) {
+      if (workspace.namespace === wsNamespace && workspace.id === wsId) {
+        return true;
+      } else {
+        return false;
+      }
+    });
+  }
+
+  public getCohort(wsNamespace: string,
+                   wsId: string,
+                   cId: string): Observable<Cohort> {
+    const observable = new Observable(observer => {
+      setTimeout(() => {
+        const workspaceMatch = this.getMatchingWorkspace(wsNamespace, wsId);
+        if (workspaceMatch !== null) {
+          observer.next(workspaceMatch.cohorts.find(function(cohort: Cohort) {
+            if (cohort.id === cId) {
+              return true;
+            }
+          }));
+          observer.complete();
+        }
+        observer.error(new Error('Error fetching a cohort. No cohort with \
```

Comment:
Thanks for the informative error! I slightly prefer
```
`Error fetching. No cohort with id: ${cId} ...`
```
string formatting in cases like this, but up to you.